Core/Src/CAN/inverter_can_diff.c

```diff
@@ -0,0 +1,44 @@
+/*
+ * inverter_diff.c
+ *
+ *  Created on: Jan 24, 2023
+ *      Author: caio
+ */
+
+#include "CAN/CAN_IDs.h"
+#include "cmsis_os.h"
+#include "util/global_instances.h"
+#include "util/global_variables.h"
+#include "util/util.h"
+
+
+static FDCAN_RxHeaderTypeDef RxHeader;
+
+// Function that gets ID's from CAN messages received from inverters and sets the flag if both are active
+void inverter_diff(void* argument) {
+
+    UNUSED(argument);
+
+    for (;;) {
+        ECU_ENABLE_BREAKPOINT_DEBUG();
+
+        uint32_t id = RxHeader.Identifier;
+
+        osMessageQueueGet(q_ids_can_inverterHandle, &id, NULL, osWaitForever);
+
+        bool right_inv_received = false;
+        bool left_inv_received  = false;
+
+        if (id == ID_RIGHT_INVERTER) {
+            right_inv_received = true;
```

Comment:
ID_RIGHT_INVERTER é o id que a ECU envia para o inversor, não que a ECU recebe do inversor que vc quer.
```suggestion
const bool right_inv_received = (id  >= 200 && id <= 203);
```
Esses ids vc acha no relatório do inversor ou no arquivo inverter_can_data_manager.h